packages/http-server-csharp/src/lib/utils.ts

```diff
@@ -587,6 +589,38 @@ export function getModelAttributes(
   return getAttributes(program, entity, cSharpName);
 }
 
+export function getModelExceptionConstructor(
+  program: Program,
+  model: Model,
+  className: string,
+): string {
+  function getDefinedStatusCode() {
+    const statusCodeProperty = new ModelInfo().filterAllProperties(program, model, (p) =>
+      isStatusCode(program, p),
+    );
+
+    if (!statusCodeProperty) return undefined;
+
+    if (statusCodeProperty.type.kind === "Number") {
+      return statusCodeProperty.type.value;
+    }
+
+    return getMinValue(program, statusCodeProperty) ?? undefined;
+  }
+
+  if (!isErrorModel(program, model)) {
+    return "";
+  }
+
+  const definedStatusCode = getDefinedStatusCode();
+  const defaultStatusCode = 400;
+
+  return definedStatusCode
```

Comment:
It seems like we also want to pass in any types in the body of the expression as the value, and that these should be strongly typed, based on the model type.

Reply:
I don't really understand this comment, do you mean on the constructor? or the class in general?